pkg/csi/csi.go

```diff
@@ -58,19 +59,37 @@ func InitializeSidecars(ver string) error {
 		return err
 	}
 
-	sv := version.MustParseGeneric(ver)
-
-	for _, image := range si {
-		v := version.MustParseGeneric(image.Version)
-		if sv.Major() == v.Major() && sv.Minor() == v.Minor() {
-			sidecarImages = &image
-			break
+	pltVersion := version.MustParseGeneric(ver)
+
+	// this helps in not parsing the version twice in loop.
+	closestVersion := pltVersion.WithMinor(0)
```

Comment:
Please initialize to `-1`. `0` creates an edge case where the z stream version is `X.0`

Reply:
1. we can't create minor version w/ -1
2. I know of X.0 but didn't change that due to my incorrect assumption
how about
`if closestVersion.Minor() == pltVersion.Minor()` to `if closestVersion.Minor() == pltVersion.Minor() && sidecarImage != nil`

Reply:
If we don't want to do an early exit this extra condition can go away